### ISSUE TITLE
[Proposal] Replace the rendering frontend with "noraesae/pen", abandoning pandoc.

### ISSUE THREAD

**Issue by simnalamburt** [junegunn/vim-xmark#11]
### TL;DR
Let's replace the rendering frontend with [noraesae/pen], abandoning pandoc.

### Content
Your [vim-xmark] is great and it had showed me how should we make a markdown previewer for vim. It works well and it's very easy to use. I've enjoyed using it for [many years].

But recently I've found [noraesae/pen], and it's UX was quite sleek. [noraesae/pen] has some advantage comparing to the [vim-xmark]:

1.  [noraesae/pen]'s dependency was much more common.

    It requires only Node.js instead of both brew and pandoc. Yes of cource, most people using macOS will install node.js using brew so effectively it just requires Node.js instread of pandoc. But most developers are highly likely to already have Node.js runtime in their computer, so I think this is a big advantage.

1.  [noraesae/pen] does not re-render entire DOM using React. So it gives user much better UX.

1.  [noraesae/pen] does not create temporary files. Instead, it uses websocket to communicate between browser and renderer.

1.  [noraesae/pen]'s github-markdown-css is updated to the more recent one.

Right now I'm using the [noraesae/pen] but I miss the ease of [vim-xmark] at the same time. I want to merge benefit the both ease of [vim-xmark] and performance of [noraesae/pen].

Let's replace the rendering frontend with [noraesae/pen] abandoning pandoc. How do you think?

CC @noraesae 

[many years]: https://github.com/junegunn/vim-xmark/pull/6
[vim-xmark]: https://github.com/junegunn/vim-xmark
[noraesae/pen]: https://github.com/noraesae/pen

**Comment by junegunn** [comment]
Hi, thanks for the suggestion. No reason to object if it doesn't make it more difficult for the users to install and use xmark. But I don't have time for this, patches are welcome.

Pen looks nice, but, correct me if I'm wrong, I presume there are alternatives to pen, like, I've seen [plenty](https://github.com/JamshedVesuna/vim-markdown-preview), [of](https://github.com/suan/vim-instant-markdown), [Vim](https://github.com/nelstrom/vim-markdown-preview), [plugins](https://github.com/shime/vim-livedown), for previewing markdown files, so I wonder what are the advantages of pen, what makes it stand out from others.
